ARSN-503: Return the kms error without warning

BourgoisMickael opened the pull request:
objectCopy and objectPutCopyPart use these functions from dataWrapper and they both catch the error to return it to the client

The KMS implementation should take care of hidding sensitive internal information

Rework error message for AWS KMS to match the output from AWS and hide some non KMS related message such as Network errors to prevent leaking sensitive information like hostnames, ip, ..

BourgoisMickael committed:
ARSN-503: Return the kms error without warning

objectCopy and objectPutCopyPart use these functions
and they both catch the error to return it to the client

The KMS implementation should take care of hidding
sensitive internal information

bert-e commented:
# Hello bourgoismickael,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/arsenal/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Request integration branches



Waiting for integration branch creation to be requested by the user.

To request integration branches, please comment on this pull request with the following command:

```
/create_integration_branches
```

Alternatively, the `/approve` and `/create_pull_requests` commands will automatically
create the integration branches.

BourgoisMickael committed:
ARSN-503: Update AWS KMS error render

Always return aws error code as is prefixed with KMS

For some error return the same error code and message
For others hide potentially sensitive message

BourgoisMickael committed:
ARSN-503: Bump version

BourgoisMickael commented:
/create_integration_branches

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/improvement/ARSN-503-kms-err` with contents from `improvement/ARSN-503-kms-err`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/improvement/ARSN-503-kms-err origin/development/8.1
 $ git merge origin/improvement/ARSN-503-kms-err
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/improvement/ARSN-503-kms-err
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.2/improvement/ARSN-503-kms-err` with contents from `w/8.1/improvement/ARSN-503-kms-err`
and `development/8.2`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.2/improvement/ARSN-503-kms-err origin/development/8.2
 $ git merge origin/w/8.1/improvement/ARSN-503-kms-err
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.2/improvement/ARSN-503-kms-err
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_integration_branches**

anurag4DSB commented on lib/network/utils.ts:
```diff
@@ -28,6 +29,61 @@ export function arsenalErrorKMIP(err: string | Error) {
+// See specific actions like CreateKey, Encrypt, Decrypt, etc. for more details.
+// Other errors not listed will return only the same error code but http status 500
+// and message will be generic and details will be in logs only.
+const allowedKmsErrors = {
```
Shouldn't we have these errors in the https://github.com/scality/Arsenal/blob/development/8.2/lib/errors/arsenalErrors.ts

BourgoisMickael replied:
They are KMS specific errors and in the S3 context the code should be prefixed with `KMS.` 

I did not want to include many new errors and duplicate them to have 2 different codes for the same error.
We should not do anything else than return them back to client that why I only encapsulated them in arsenal error

anurag4DSB replied:
THere is no duplication see [this](https://github.com/scality/Arsenal/blob/development/8.2/lib/errors/arsenalErrors.ts#L1076)
I think its still worth adding all errors there like the KMS one already present

BourgoisMickael replied:
I've reworked kms errors in arsenal [97d9bfe](https://github.com/scality/Arsenal/pull/2432/commits/97d9bfefb6896ca839e88cdc53c53328a7fc0f99)

anurag4DSB reviewed (approved):
I am approving the PR, your call on my comment

BourgoisMickael committed:
ARSN-503: Test KMS arsenalError

bert-e commented:
# History mismatch



Merge commit #98eb4e01a24fb82680d959c0df2e07fc5aa2cccd on the integration branch
`w/8.1/improvement/ARSN-503-kms-err` is merging a branch which is neither the current
branch `improvement/ARSN-503-kms-err` nor the development branch
`development/8.1`.

It is likely due to a rebase of the branch `improvement/ARSN-503-kms-err` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_integration_branches**

BourgoisMickael commented:
/approve

bert-e commented:
# Build failed



The [build](https://github.com/scality/Arsenal/actions?query=branch%3Aw/8.2/improvement/ARSN-503-kms-err) for [commit](https://github.com/scality/Arsenal/commit/dfaa5365ef688a904457650e05f98e874e4e9aec) did not succeed in branch [w/8.2/improvement/ARSN-503-kms-err](https://github.com/scality/arsenal/compare/development/8.2...w/8.2/improvement/ARSN-503-kms-err)



*The following options are set:* **approve, create_integration_branches**

bert-e commented:
# History mismatch



Merge commit #a8521c66c132eca9ca13730bf950ef9647f814c5 on the integration branch
`w/8.1/improvement/ARSN-503-kms-err` is merging a branch which is neither the current
branch `improvement/ARSN-503-kms-err` nor the development branch
`development/8.1`.

It is likely due to a rebase of the branch `improvement/ARSN-503-kms-err` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve, create_integration_branches**

bert-e commented:
# Build failed



The [build](https://github.com/scality/Arsenal/actions?query=branch%3Aw/8.2/improvement/ARSN-503-kms-err) for [commit](https://github.com/scality/Arsenal/commit/c10676b4f3bfa0312e74c9934806c13a0dcb0a2b) did not succeed in branch [w/8.2/improvement/ARSN-503-kms-err](https://github.com/scality/arsenal/compare/development/8.2...w/8.2/improvement/ARSN-503-kms-err)



*The following options are set:* **approve, create_integration_branches**

bert-e commented:
# History mismatch



Merge commit #83dab2404a431765c9470959e2db74e117f4d0eb on the integration branch
`w/8.1/improvement/ARSN-503-kms-err` is merging a branch which is neither the current
branch `improvement/ARSN-503-kms-err` nor the development branch
`development/8.1`.

It is likely due to a rebase of the branch `improvement/ARSN-503-kms-err` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve, create_integration_branches**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/7.70`

* :heavy_check_mark: `development/8.1`

* :heavy_check_mark: `development/8.2`



The following branches have **NOT** changed:

* `development/6.4`
* `development/7.10`
* `development/7.4`



Please check the status of the associated issue ARSN-503.

Goodbye bourgoismickael.




*The following options are set:* **approve, create_integration_branches**
